### ISSUE TITLE
why InferenceBatchSoftmax only on testing but not training?

### ISSUE THREAD

**Issue by CuriousDeepLearner** [SeanNaren/deepspeech.pytorch#361]
Can anyone here explain to me why we only do softmax on testing but not training please?
```python
class InferenceBatchSoftmax(nn.Module): 
     def forward(self, input_):
           if not self.training:
                return F.softmax(input_, dim=-1)
          else:
               return input_
```
I try to run with these above and the CTC loss return is negatif, but when I do softmax on training, it seems normal. Did i miss something here? Thanks
